src/storages/pluggable/index.ts

```diff
@@ -62,6 +67,16 @@ export function PluggableStorage(options: PluggableStorageOptions): IStorageAsyn
       // Disconnect the underlying storage, to release its resources (such as open files, database connections, etc).
       destroy() {
         return wrapper.close();
+      },
+
+      // emits SDK_READY event on shared clients and returns a reference to the storage
+      shared(_, onReadyCb) {
+        wrapperConnect(wrapper, onReadyCb);
+        return {
+          ...this,
+          // no-op destroy, to close the wrapper only when the main client is destroyed
+          destroy() { }
```

Comment:
Is the assumption here is that given the nature of the client side API, we won't have a possible implementation where you might want to free up resources per "user"?

Reply:
Nico, it is done this way to avoid calling the `wrapper.close` if a shared clients destroy method is called.